src/solvers/interior_point_Newton.jl

```diff
@@ -0,0 +1,382 @@
+    grad_g=nothing,
+    grad_h=nothing,
+    Hess_g=nothing,
+    Hess_h=nothing,
+    inequality_constrains::Union{Integer,Nothing}=nothing,
+    equality_constrains::Union{Nothing,Integer}=nothing,
+    kwargs...,
+)
+    q = copy(M, p)
+    cmo = ConstrainedManifoldObjective(
+        f,
+        grad_f,
+        g,
+        grad_g,
+        h,
+        grad_h;
+        hess_f=Hess_f,
+        hess_g=Hess_g,
+        hess_h=Hess_h,
+        evaluation=evaluation,
+        inequality_constrains=inequality_constrains,
+        equality_constrains=equality_constrains,
+        M=M,
+        p=p,
+    )
+    return interior_point_Newton!(M, cmo, q; evaluation=evaluation, kwargs...)
+end
+function interior_point_Newton(
+    M::AbstractManifold, cmo::O, p; kwargs...
+) where {O<:Union{ConstrainedManifoldObjective,AbstractDecoratedManifoldObjective}}
+    q = copy(M, p)
+    return interior_point_Newton!(M, cmo, q; kwargs...)
+end
+
+@doc "$(_doc_IPN)"
+interior_point_Newton!(M::AbstractManifold, args...; kwargs...)
+
+function interior_point_Newton!(
+    M::AbstractManifold,
+    f,
+    grad_f,
+    Hess_f,
+    p;
+    evaluation::AbstractEvaluationType=AllocatingEvaluation(),
+    g=nothing,
+    h=nothing,
+    grad_g=nothing,
+    grad_h=nothing,
+    Hess_g=nothing,
+    Hess_h=nothing,
+    inequality_constrains=nothing,
+    equality_constrains=nothing,
+    kwargs...,
+)
+    cmo = ConstrainedManifoldObjective(
+        f,
+        grad_f,
+        g,
+        grad_g,
+        h,
+        grad_h;
+        hess_f=Hess_f,
+        hess_g=Hess_g,
+        hess_h=Hess_h,
+        evaluation=evaluation,
+        equality_constrains=equality_constrains,
+        inequality_constrains=inequality_constrains,
+        M=M,
+        p=p,
+    )
+    dcmo = decorate_objective!(M, cmo; kwargs...)
+    return interior_point_Newton!(
+        M,
+        dcmo,
+        p;
+        evaluation=evaluation,
+        equality_constrains=equality_constrains,
+        inequality_constrains=inequality_constrains,
+        kwargs...,
+    )
+end
+function interior_point_Newton!(
+    M::AbstractManifold,
+    cmo::O,
+    p;
+    evaluation::AbstractEvaluationType=AllocatingEvaluation(),
+    X=get_gradient(M, cmo, p),
+    μ::Vector=ones(length(get_inequality_constraint(M, cmo, p, :))),
+    Y=zero(μ),
+    λ::Vector=zeros(length(get_equality_constraint(M, cmo, p, :))),
+    Z=zero(λ),
+    s=copy(μ),
+    W=zero(s),
+    ρ=μ's / length(μ),
+    σ=calculate_σ(M, cmo, p, μ, λ, s),
+    retraction_method::AbstractRetractionMethod=default_retraction_method(M, typeof(p)),
+    sub_kwargs=(;),
+    vector_space=Rn,
+    #γ=0.9,
+    centrality_condition=missing, #InteriorPointCentralityCondition(cmo, γ, zero(γ), zero(γ)),
+    step_objective=ManifoldGradientObjective(
+        KKTVectorFieldNormSq(cmo), KKTVectorFieldNormSqGradient(cmo); evaluation=evaluation
+    ),
+    _step_M=M × vector_space(length(μ)) × vector_space(length(λ)) × vector_space(length(s)),
```

Comment:
```suggestion
    _step_M::AbstractManifold=M × vector_space(length(μ)) × vector_space(length(λ)) × vector_space(length(s)),
```

Reply:
With the `_` upfront I expect this to be an internal keyword only anyways (should not be set by the user, makes a few other keywords shorter to specify the default), but sure we can still type them.